dateutil/rrule.py

```diff
@@ -343,7 +345,24 @@ def __init__(self, freq, dtstart=None,
             dtstart = datetime.datetime.fromordinal(dtstart.toordinal())
         else:
             dtstart = dtstart.replace(microsecond=0)
+
+
         self._dtstart = dtstart
+
+        # <TIMEZONE HACK>
+
+        # if tzid is set, force the timezone on dtstart
+        # this is to support generating recurrences from events stored in PostgreSQL, 
+        # which converts all timezones to UTC on storage
+        if tzid:
+          tzid = gettz(tzid)
+          if dtstart.tzinfo:
+            dtstart = dtstart.astimezone(tz)
```

Comment:
My mistake; I meant to merge into the fork rather than back into the main repository. Going to close the PR. For clarity, this is a fix that is meant to force setting a timezone in a Postgres-stack where all timestamps are stored as UTC regardless of input timezone.

As you said, it is possible to do so outside of dateutil, but doing so would require a major sweep over our code and so I decided to push it into a fork of dateutil.rrule, resulting in a minimal change.

Sorry about the confusion!